web/modules/custom/dpl_logging/dpl_logging.module

```diff
@@ -0,0 +1,113 @@
+<?php
+
+use Drupal\Component\Utility\DiffArray;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\user\Entity\User;
+
+/**
+ * Implements hook_ENTITY_TYPE_presave().
+ *
+ * When updating a user, we want to log the changes done - this is part of
+ * a security policy by DDF.
+ * For most fields, the old and new value gets logged, but others may be
+ * ignored, or declared on a meta-level (such as password).
+ */
+function dpl_logging_user_presave(User $user): void {
+  $original_user = $user->original;
+
+  if (!($original_user instanceof User)) {
+    return;
+  }
+
+  $original_values = $original_user->toArray();
+  $values = $user->toArray();
+  $updates = DiffArray::diffAssocRecursive($values, $original_values);
+
+  // Some fields are so meta that we do not need to log their changes.
+  $ignored_updates = ['changed', 'metatag', 'field_password_expiration'];
+  $diffs = [];
+
+  foreach ($updates as $key => $value) {
+    if (in_array($key, $ignored_updates)) {
+      continue;
+    }
+
+    if ($key === 'roles') {
+      $original_roles = $original_user->getRoles();
+      $new_roles = $user->getRoles();
+
+      // This results in the machine names of the roles.
+      // We could also do a look up, to find the translated labels also,
+      // but it is probably overkill for what this log will be used for.
+      $added_roles = array_diff($new_roles, $original_roles);
+      $removed_roles = array_diff($original_roles, $new_roles);
```

Comment:
I agree with this. I would argue the same this for the fields but I see you went with both labels and keys there.